pkg/cluster/inventory.go

```diff
@@ -350,10 +375,121 @@ func (i *DefaultInventory) latestCluster(cluster string) (*model.ClusterEntity,
+	if err != nil {
+		return nil, err
+	}
+	idColName, err := statusColHandler.ColumnName("ID")
+	if err != nil {
+		return nil, err
+	}
+	clusterColName, err := statusColHandler.ColumnName("Cluster")
+	if err != nil {
+		return nil, err
+	}
+	clusterVersionColName, err := statusColHandler.ColumnName("ClusterVersion")
+	if err != nil {
+		return nil, err
+	}
+	configVersionColName, err := statusColHandler.ColumnName("ConfigVersion")
+	if err != nil {
+		return nil, err
+	}
+	statusColName, err := statusColHandler.ColumnName("Status")
+	if err != nil {
+		return nil, err
+	}
+	createdColName, err := statusColHandler.ColumnName("Created")
+	if err != nil {
+		return nil, err
+	}
+
+	//get cluster configurations of all "not-ready" clusters
+	q, err := db.NewQuery(i.Conn, &model.ClusterConfigurationEntity{})
+	if err != nil {
+		return nil, err
+	}
+
+	/*
+		select config_version from inventory_cluster_config_statuses where id in (
+			select maxid from (
+				select max(cluster_version) as maxcfg, max(id) as maxid from inventory_cluster_config_statuses group by cluster
+			) as maxid
+		) and (status in ('xxx', 'yyy') or (status = 'ready' AND created >=  NOW() - INTERVAL '12345 SECOND'))
+	*/
+	statusFilter := fmt.Sprintf("%s IN ('%s')", statusColName, strings.Join(i.stausesToStrings(statuses), "','"))
+	if reconcileInterval.Seconds() > 0 {
+		switch i.Conn.Type() {
+		case db.Postgres:
+			statusFilter = fmt.Sprintf(`(
+				%s
+				OR (
+					%s = '%s' AND %s >= NOW() - INTERVAL '%.0f SECOND')
+				)`,
+				statusFilter, statusColName, model.Ready, createdColName, reconcileInterval.Seconds())
+		case db.SQLite:
+			statusFilter = fmt.Sprintf(`(
+				%s
+				OR (
+					%s = '%s' AND %s >= DATETIME('now', '-%.0f SECONDS'))
+				)`,
+				statusFilter, statusColName, model.Ready, createdColName, reconcileInterval.Seconds())
+		default:
+			return nil, fmt.Errorf("Database type '%s' is not supported by this method", i.Conn.Type())
+		}
+	}
+
+	clusterConfigs, err := q.Select().
+		WhereIn("Version",
+			fmt.Sprintf(`SELECT %s FROM %s WHERE %s IN (
+							SELECT maxid FROM (
+								SELECT MAX(%s) AS maxcfg, MAX(%s) AS maxid FROM %s GROUP BY %s
+							) AS maxid
+						) AND %s`,
+				configVersionColName, clusterStatus.Table(), idColName,
+				clusterVersionColName, idColName, clusterStatus.Table(), clusterColName,
+				statusFilter)).
+		Where(map[string]interface{}{
+			"Deleted": false,
+		}).
+		GetMany()
+	if err != nil {
+		return nil, err
+	}
+
+	//retreive clusters which require a reconciliation
+	result := []*State{}
+	for _, clusterConfig := range clusterConfigs {
+		clusterConfigEntity := clusterConfig.(*model.ClusterConfigurationEntity)
+		state, err := i.Get(clusterConfigEntity.Cluster, clusterConfigEntity.Version)
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, state)
+	}
+
+	return result, nil
+}
+
+func (i *DefaultInventory) stausesToStrings(statuses []model.Status) []string {
```

Comment:
There is a typo in the function name. We can even make it `convertStatusesToStrings` if you like.